modules/dex/src/lib.rs

```diff
@@ -490,6 +520,98 @@ impl<T: Trait> Module<T> {
 			)
 		}
 	}
+
+	pub fn deposit_calculate_interest(currency_id: CurrencyIdOf<T>, who: &T::AccountId, share_amount: T::Share) {
+		let total_shares = Self::total_shares(currency_id);
+		if total_shares.is_zero() {
+			return;
+		}
+		let proportion = Ratio::from_rational(share_amount, total_shares);
+		let total_interest = Self::total_interest(currency_id);
+		if total_interest.is_zero() {
+			return;
+		}
+		let new_debits = proportion.saturating_mul_int(&total_interest);
+		<Debits<T>>::mutate(currency_id, &who, |debits| {
+			*debits = debits.saturating_add(new_debits);
+		});
+		<TotalDebits<T>>::mutate(currency_id, |total_debits| {
+			*total_debits = total_debits.saturating_add(new_debits);
+		});
+		<TotalInterest<T>>::mutate(currency_id, |interest| {
+			*interest = interest.saturating_add(new_debits);
+		});
+	}
+
+	fn withdraw_calculate_interest(currency_id: CurrencyIdOf<T>, who: &T::AccountId, share_amount: T::Share) {
+		Self::claim_interest(currency_id, who.clone());
+		let shares = Self::shares(currency_id, who.clone());
+		let debits = Self::debits(currency_id, who.clone());
+		let proportion = Ratio::from_rational(share_amount, Self::total_shares(currency_id));
+		let remove_debits = Ratio::from_rational(share_amount, shares).saturating_mul_int(&debits);
+		let (_, base_currency_pool) = Self::liquidity_pool(currency_id);
+		let withdrawn_interest: T::Share = proportion.saturating_mul_int(&base_currency_pool).into();
+
+		<Debits<T>>::mutate(currency_id, &who, |debits| {
+			*debits = debits.saturating_sub(remove_debits);
+		});
+		<TotalDebits<T>>::mutate(currency_id, |total_debits| {
+			*total_debits = total_debits.saturating_sub(remove_debits);
+		});
+		<TotalInterest<T>>::mutate(currency_id, |interest| {
+			*interest = interest.saturating_sub(proportion.saturating_mul_int(interest));
+		});
+		<Withdrawn<T>>::mutate(currency_id, &who, |withdrawn| {
+			*withdrawn = withdrawn.saturating_add(withdrawn_interest);
+		});
+	}
+
+	fn claim_interest(currency_id: CurrencyIdOf<T>, who: T::AccountId) {
+		let shares = Self::shares(currency_id, who.clone());
+		let debits = Self::debits(currency_id, who.clone());
+		let proportion = Ratio::from_rational(shares, Self::total_shares(currency_id));
+		let total_interest = Self::total_interest(currency_id);
+		let withdrawn_interest = proportion.saturating_mul_int(&total_interest).saturating_sub(debits);
+		<Debits<T>>::mutate(currency_id, &who, |debits| {
+			*debits = debits.saturating_add(withdrawn_interest);
+		});
+		<TotalDebits<T>>::mutate(currency_id, |debits| {
+			*debits = debits.saturating_add(withdrawn_interest);
+		});
+		<Withdrawn<T>>::mutate(currency_id, &who, |withdrawn| {
```

Comment:
should use `T::Currency::deposit` to mint aUSD for user

Reply:
deposit aUSD without cdp backed, must increase debit to cdp treasury at the same time.
Extend trait `CDPTreasury`, add `deposit_unbacked_debit` function to do it.